src/commands.c

```diff
@@ -348,22 +375,52 @@ RRStatus CommandSpecTableSetC(CommandSpecTable *cmd_spec_table, void *key, size_
 /* Look up the specified command in the command spec table and return the
  * CommandSpec associated with it, or NULL.
  */
-const CommandSpec *CommandSpecTableGet(CommandSpecTable *cmd_spec_table, const RedisModuleString *cmd)
+int CommandSpecTableGetFlags(CommandSpecTable *cmd_spec_table, RedisModuleDict *sub_command_tables, const RedisModuleString *cmd, const RedisModuleString *subcmd)
 {
-    return getOrCreateCommandSpec(cmd_spec_table, cmd, false);
+    CommandSpec *cs = getOrCreateCommandSpec(cmd_spec_table, cmd, false);
+
+    if (!cs) {
+        return -1;
+    }
+
+    int flags = cs->flags;
+
+    if (!(cs->flags & CMD_SPEC_SUBCOMMAND) || subcmd == NULL) {
+        return flags;
+    }
+
+    cmd_spec_table = RedisModule_DictGetC(sub_command_tables, cs->name, strlen(cs->name), NULL);
+    if (cmd_spec_table == NULL) {
+        return flags;
+    }
+
+    /* flags are now for subcommand, so mask it out of flags */
+    flags &= ~CMD_SPEC_SUBCOMMAND;
```

Comment:
What is the difference between this case and the one above? in both cases we get a command with a subcommand.

Reply:
```
    if (!(cs->flags & CMD_SPEC_SUBCOMMAND) || subcmd == NULL) {
        return flags;
    }
```

either the command we got is not a subcommand container or we didn't get a subcommand for it, therefore return its flags as is.  This will be the very common case

```
    CommandSpecTable *sub_cmd_spec_table = RedisModule_DictGetC(sub_command_tables, cs->name, strlen(cs->name), NULL);
    if (sub_cmd_spec_table == NULL) {
        return flags;
    }
```

it is a subcommand container, and has a subcommand, but, we didn't create a subcommand table for it, therefore return its flags as is (though perhaps can return it masked, as it is a subcommand, that would make sense to me

```
    /* flags are now for subcommand, so mask it out of flags */
    flags &= ~CMD_SPEC_SUBCOMMAND;
```

for subcommands (i.e. my note above), we mask out the the SUBCOMMND part, the rest is the default values returned for subcommands not specified

```
    CommandSpec *sub_cs = getOrCreateCommandSpec(sub_cmd_spec_table, subcmd, false);
    if (sub_cs == NULL) {
        return flags;
    }
```

look up the subcommand, if its not specified, returned default (now masked) value.

```
    return sub_cs->flags;
```
otherwise return value defined for the subcommand.

Reply:
added the above notes as comments in the func